File: solutions/ruby/eliuds-eggs/1/eliuds_eggs.rb
class EliudsEggs
  # Cambiamos el nombre de 'count' a 'egg_count'
  def self.egg_count(number)
    # Tu lógica original es excelente:
    # 1. Convertimos a binario: 89 -> "1011001"
    # 2. Separamos caracteres: ["1", "0", "1", "1", "0", "0", "1"]
    # 3. Contamos los unos: 4
    number.to_s(2).chars.count('1')
  end
end

File: solutions/java/eliuds-eggs/1/src/main/java/EliudsEggs.java
public class EliudsEggs {
    public int eggCount(int number) {
        int count = 0;
        while (number > 0) {
            // Check if the least significant bit is 1
            if ((number & 1) == 1) {
                count++;
            }
            // Unsigned right shift to check the next bit
            number >>>= 1;
        }
        return count;
    }
}